README.md

```diff
@@ -609,27 +603,37 @@ from({ hello: [ 'world' ]}).hello[0].concat('!!!').valueOf();
 // { hello: [ 'world!!!' ]}
 ```
 
-## map(fn: tree => tree, microstate: Microstate): Microstate
+## map(fn, microstate): Microstate
 
-The `map` function is used to perform operations on the tree that are used to build a new Microstate. It expects a function and a Microstate and returns a Microstate. This function accepts a mapping function which will receive the Microstate's tree. The mapping function is expected to return a tree. The tree that is returned from the mapping will be used to generate the microstate that's returned by the map operation.
+The `map` function invokes the function for each microstate in an array microstate. It usually used to map over an array of microstate and return an array of components. The mapping function will receive each microstate in the array. You can invoke transitions on each microstate as you would usually.
 
-This is most frequently used go derive a microstate with middleware installed.
+```js
+let numbers = create([Number], [1, 2, 3, 4]);
+
+<ul>
+  {map(number => (
+    <li onClick={() => number.increment()}>{number.state}</li>
+  ), numbers)}
+</ul>
```

Comment:
How would this pattern work in Ember, would `{{#each nubmer as |number|}}` be enough, or does `map` do something special?

Reply:
You should not need any mapping function as long as the microstate list object (array like) is compatible with the iterating mechanism in Glimmer. We might need to make a little tweak to microstates to ensure this compatibility. We will make it work with Ember without anything special. You'll be able to just use `each` without doing anything special.

Reply:
I'm not sure @cowboyd. I'll give this a test.